lib/fs.js

```diff
@@ -1112,14 +1114,14 @@ function writeAll(fd, buffer, offset, length, position, callback) {
 }
 
 fs.writeFile = function(path, data, options, callback) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
+  callback = maybeCallback(arguments[arguments.length - 1]);
```

Comment:
@thefourtheye 
> but it still says the function is not optimized for both the cases.

Have you enabled `'use strict'` for the file or for the function?

```javascript
function unoptimized0(a, b) {
	b = arguments[arguments.length - 1] + 1;
	return b * 1;
}
function unoptimized1(a, b) {
	var b = arguments[arguments.length - 1] + 1;
	return b * 1;
}
function optimized0(a, b) {
	'use strict';
	b = arguments[arguments.length - 1] + 1;
	return b * 1;
}
function unoptimized0(a, b) {
	'use strict';
	var b = arguments[arguments.length - 1] + 1;
	return b * 1;
}
```